app/src/models/branch.ts

```diff
@@ -68,6 +68,20 @@ export class Branch {
     return pieces[1]
   }
 
+  /** The name of remote for a remote branch. If local, will return null. */
+  public get remoteName(): string | null {
+    if (this.type === BranchType.Local) {
+      return null
+    }
+
+    const pieces = this.name.match(/(.*?)\/.*/)
+    if (!pieces || pieces.length < 2) {
+      return null
+    }
+
+    return pieces[1]
```

Comment:
To play with this I made a branch called `origin/test` along with my `test` and current code doesn't even seem to know what to do with it to the point where it errors with popup. 

I will make a PR to introduce the conical ref, and I think there may already be an underlying issue with how local names are parsed... thus this might be useful for both.... but needs further investigation.

Reply:
PR https://github.com/desktop/desktop/pull/11490 to resolve this